hera_qm/xrfi.py

```diff
@@ -276,46 +277,28 @@ def detrend_dpss_filt(data, noise, freqs, filter_centers, filter_half_widths, wg
     if len(eigenval_cutoff) == 1 and len(filter_centers) > 1:
         eigenval_cutoff = len(filter_centers) * eigenval_cutoff
 
-    if wgts is None:
+    if flags is None:
         wgts = np.ones_like(data)
-
+    elif flags is not None and flags.dtype != bool:
+        raise TypeError('Input flag array must be type bool')
     else:
-        wgts = deepcopy(wgts)
-
-    # Compute first-pass model and residuals
-    model, _, _ = dspec.fourier_filter(
-        freqs, data, wgts, filter_centers, filter_half_widths, mode=mode,
-        suppression_factors=suppression_factors, eigenval_cutoff=eigenval_cutoff,
-        cache=cache
-    )
-    res = data - model
-
-    # Compute modified z-score to set preliminary flags
-    sigma = np.median(
-        np.abs(res - np.median(res, keepdims=True, axis=1)), axis=1, keepdims=True,
-    )
-    sigma /= 0.6745
-
-    # If channels are flag, set weights to zero. Otherwise use inversion noise-variance weighting
-    wgts = np.where(np.abs(res) < nsig * sigma, noise ** -2, 0)
-
-    # Recompute model with first pass weights
+        wgts = np.array(np.logical_not(flags), dtype=np.float64)
+    
+    # Compute model and residuals
     model, _, _ = dspec.fourier_filter(
         freqs, data, wgts, filter_centers, filter_half_widths, mode=mode,
-        suppression_factors=suppression_factors, eigenval_cutoff=eigenval_cutoff,
-        cache=cache
+        suppression_factors=suppression_factors, eigenval_cutoff=eigenval_cutoff, cache=cache,
     )
-    res = data - model
-
+    res = data - model    
+    
     # Use smooth model to noise standard deviation without RFI
     sigma = np.abs(model) * (noise / data)
-
+    
     # Determine weights
-    wgts = np.where(np.abs(res) <= nsig * sigma, 1, 0)
-    return wgts
+    return np.where(res > sigma * nsig, True, False)
 
 
-def detrend_channel_diff_filt(data, noise, nsig=6, kernel_widths=[1, 2, 4], wgts=None):
+def channel_diff_flagger(data, noise, nsig=6, kernel_widths=[3, 4, 5], flags=None):
```

Comment:
You probably want to update the docstring to clarify that it returns flags, not weights

Reply:
Updated this